profiler/hotapp/Dockerfile

```diff
@@ -1,12 +1,10 @@
 FROM golang:alpine
```

Comment:
I think we should try to pin the version of Go and alpine. Otherwise we'll run into build issues again next time we have to build an image

Reply:
Agreed, but I did not want to change too much in this PR. 
I can address this in another one once I am sure that these changes have the intended effect.

Reply:
Pinned version to `1.20.7-alpine`